src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -648,6 +650,89 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^Now: {}}` */
+class NowGenerator : public Generator<bsoncxx::types::b_date> {
+public:
+    NowGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_date evaluate() override {
+        return bsoncxx::types::b_date{std::chrono::system_clock::now()};
+    }
+};
+
+// The 2 formats also cover "%Y-%m-%d", timezones are not parsed,
+// %z, %Q are output only https://www.boost.org/doc/libs/1_75_0/doc/html/date_time/date_time_io.html.
+const std::locale formats[] = {
+    std::locale(std::locale::classic(),new boost::posix_time::time_input_facet("%Y-%m-%dT%H:%M:%S.%F")),
+    std::locale(std::locale::classic(),new boost::posix_time::time_input_facet("%Y-%m-%d %H:%M:%S.%F")),
```

Comment:
1. can these be smartptrs? `std::make_unique<time_input_face>("%Y--")`. Seeing "naked new" anywhere makes my spidey-senses perk-up. If that's not possible, plz put a comment saying as such. Not a huge deal since this is statically-allocated storage but still good to ack when we have to break modern-isms.
2. I *think* the formatter would put a space after the comma. Maybe you did already but plz double-check you ran `clang-format` prior to merge.